### ISSUE TITLE
Error while trying to use alidock (Docker method): Cannot continue: the host initialization script failed, check ~/alidock/.alidock-host.log

### ISSUE THREAD

**Issue by danielleturvill** [alidock/alidock#143]

**Comment by danielleturvill** [comment]
Docker version is 19.03.1


drwxr-xr-x@ 14 danielleturvill  staff   448 28 Aug 15:20 /Users/danielleturvill/alidock/.
drwxr-xr-x+ 54 danielleturvill  staff  1728 28 Aug 15:04 /Users/danielleturvill/alidock/..
-rw-r--r--   1 danielleturvill  staff   451 28 Aug 15:18 /Users/danielleturvill/alidock/.alidock-host.log
-rwx------   1 danielleturvill  staff  1161 28 Aug 15:18 /Users/danielleturvill/alidock/.alidock-init-host.sh
-rwx------   1 danielleturvill  staff  4317 28 Aug 15:18 /Users/danielleturvill/alidock/.alidock-init.sh
drwxr-xr-x@  4 danielleturvill  staff   128 28 Aug 15:20 /Users/danielleturvill/alidock/.alidock-ssh
-rw-r--r--   1 danielleturvill  staff  2173 28 Aug 15:20 /Users/danielleturvill/alidock/.alidock.log
-rw-r--r--   1 danielleturvill  staff    10 28 Aug 15:05 /Users/danielleturvill/alidock/.alidock_docker_check
-rw-r--r--   1 danielleturvill  staff    10 28 Aug 15:04 /Users/danielleturvill/alidock/.alidock_pip_check
-rw-r--r--@  1 danielleturvill  staff    70 28 Aug 15:20 /Users/danielleturvill/alidock/.bash_profile
-rw-r--r--@  1 danielleturvill  staff    50 28 Aug 15:20 /Users/danielleturvill/alidock/.bashrc
-rw-r--r--@  1 danielleturvill  staff     0 28 Aug 15:20 /Users/danielleturvill/alidock/.hushlogin
lrwxr-xr-x   1 danielleturvill  staff    11 28 Aug 15:07 /Users/danielleturvill/alidock/.sw -> .sw.noindex
drwxr-xr-x@  2 danielleturvill  staff    64 28 Aug 15:07 /Users/danielleturvill/alidock/.sw.noindex

**Comment by mconcas** [comment]
Hello, thanks for posting.
Could you please send the content of:

- /Users/danielleturvill/alidock/.alidock-host.log
- /Users/danielleturvill/alidock/.alidock-init-host.sh
- /Users/danielleturvill/alidock/.alidock.log

Cheers,
Matteo

**Comment by danielleturvill** [comment]
Hi Matteo,

It seems it may be a Python problem but I can't find a clear solution. 
Here are the contents you requested (in order):

**1.**
```
+ cd /
+ [[ -d /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -L /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw.noindex ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw ]]
+ tmutil addexclusion /Users/danielleturvill/alidock/.sw.noindex
+ xattr /Users/danielleturvill/alidock/.sw.noindex
  File "/usr/bin/xattr", line 31
    continue
           ^
TabError: inconsistent use of tabs and spaces in indentation
```


**2.**
```
# macOS: exclude aliBuild directory from Spotlight searches
# This also migrates an existing .sw directory to .sw.noindex and creates proper
# symlinks. The approach is conservative: it won't destroy any custom data
if [[   -d "/Users/danielleturvill/alidock/.sw" && \
      ! -L "/Users/danielleturvill/alidock/.sw" && \
      ! -e "/Users/danielleturvill/alidock/.sw.noindex" ]]; then
  # Migrate old .sw directory to .sw.noindex
  mv -v "/Users/danielleturvill/alidock/.sw" "/Users/danielleturvill/alidock/.s$
fi
if [[ ! -e "/Users/danielleturvill/alidock/.sw.noindex" ]]; then
  mkdir -p "/Users/danielleturvill/alidock/.sw.noindex"
fi
if [[ ! -e "/Users/danielleturvill/alidock/.sw" ]]; then
  ln -nfs .sw.noindex "/Users/danielleturvill/alidock/.sw"
fi

# macOS: exclude aliBuild directory from Time Machine backups
tmutil addexclusion "/Users/danielleturvill/alidock/.sw.noindex"
xattr "/Users/danielleturvill/alidock/.sw.noindex"
```




**3.**
There is no content. 


The alternative error I get shows up as (w.r.t. the alidock-host.log):

```
+ cd /
+ [[ -d /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -L /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw.noindex ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw ]]
+ tmutil addexclusion /Users/danielleturvill/alidock/.sw.noindex
+ xattr /Users/danielleturvill/alidock/.sw.noindex
python version 2.7.16 can't run /usr/bin/xattr.  Try the alternative(s):

(Error: no alternatives found)
```




Thank you so much for your time.

Many thanks,
Danielle.

**Comment by danielleturvill** [comment]
Hi Matteo,

It seems it may be a Python problem but I can't find a clear solution. 
Here are the contents you requested (in order):

**1.**
+ cd /
+ [[ -d /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -L /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw.noindex ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw ]]
+ tmutil addexclusion /Users/danielleturvill/alidock/.sw.noindex
+ xattr /Users/danielleturvill/alidock/.sw.noindex
  File "/usr/bin/xattr", line 31
    continue
           ^
TabError: inconsistent use of tabs and spaces in indentation


**2.**
# macOS: exclude aliBuild directory from Spotlight searches
# This also migrates an existing .sw directory to .sw.noindex and creates proper
# symlinks. The approach is conservative: it won't destroy any custom data
if [[   -d "/Users/danielleturvill/alidock/.sw" && \
      ! -L "/Users/danielleturvill/alidock/.sw" && \
      ! -e "/Users/danielleturvill/alidock/.sw.noindex" ]]; then
  # Migrate old .sw directory to .sw.noindex
  mv -v "/Users/danielleturvill/alidock/.sw" "/Users/danielleturvill/alidock/.s$
fi
if [[ ! -e "/Users/danielleturvill/alidock/.sw.noindex" ]]; then
  mkdir -p "/Users/danielleturvill/alidock/.sw.noindex"
fi
if [[ ! -e "/Users/danielleturvill/alidock/.sw" ]]; then
  ln -nfs .sw.noindex "/Users/danielleturvill/alidock/.sw"
fi

# macOS: exclude aliBuild directory from Time Machine backups
tmutil addexclusion "/Users/danielleturvill/alidock/.sw.noindex"
xattr "/Users/danielleturvill/alidock/.sw.noindex"




**3.**
There is no content. 


The alternative error I get shows up as (w.r.t. the alidock-host.log):

+ cd /
+ [[ -d /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -L /Users/danielleturvill/alidock/.sw ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw.noindex ]]
+ [[ ! -e /Users/danielleturvill/alidock/.sw ]]
+ tmutil addexclusion /Users/danielleturvill/alidock/.sw.noindex
+ xattr /Users/danielleturvill/alidock/.sw.noindex
python version 2.7.16 can't run /usr/bin/xattr.  Try the alternative(s):

(Error: no alternatives found)




Thank you so much for your time.

Many thanks,
Danielle.

**Comment by dberzano** [comment]
Hi Danielle, sorry for overlooking the issue. I'd start with opening a shell, and checking some basic facts.

a) `xattr` is a basic MacOS command. It is available with every MacOS version. Incidentally, it's written in Python and it's supposed to run with the default Python version shipped with MacOS. Did you install any custom Python version that may impair the use of `xattr`? What happens if you just type `xattr` in your terminal?

b) Do you have Python 3 installed on your Mac by chance? Please run (and post the output):

```bash
type python
type python3
type pip
type pip3
```

In case you don't have Python 3, please install [Homebrew](https://brew.sh/), and, once installed, please run `brew install python`.

c) Please reinstall alidock. The automatic installer should give the following output:

```
Checking if your Docker installation works
Creating an environment for alidock using Python 3.7.3 (python3) and virtualenv 16.2.0
Installing alidock under /Users/volpe/.virtualenvs/alidock
Installed: alidock 1.4.0
You can use alidock by simply typing:
  alidock
```

Please check, in particular, the Python version, and report it.

**Comment by dturvill** [comment]
Hello Dario,

No need to apologise! I wiped my Mac and reinstalled everything I needed. The problem was resolved after doing this. It is most likely that there was a clash with Python installations.

Thank you for your post. 
Just in case it is helpful for you to know, I had carried out the steps you've outlined. I cannot remember what was output when I typed xattr. However, I did have Python 3 and had run the homebrew command. I always got the output of c) but whenever I tried to execute alidock thereafter, it would fail claiming that the initialization script failed.

Many thanks,
Danielle.
